src/mixins/size.js

```diff
@@ -19,25 +35,27 @@ export const size = {
 
   methods: {
     updateSizeObservation() {
-      if (this.shouldMeasureSize) {
-        if (this.$el && !this.cancelSizeListener) {
-          this.cancelSizeListener = watchElementSize(this.$el, size => {
-            if (!size) return;
-
-            if (
-              !this.size ||
-              this.size.width !== size.width ||
-              this.size.height !== size.height
-            ) {
-              this.size = size;
-            }
-          });
-        }
-      } else {
-        if (this.cancelSizeListener) {
-          this.cancelSizeListener();
-        }
+      if (!this.responsive) {
+        this.cancelSizeListener && this.cancelSizeListener()
+        return
       }
+
+      const isElementRendered = !!this.$el
+
+      if (!isElementRendered || this.cancelSizeListener) return
+
+      this.cancelSizeListener = watchElementSize(this.$el, newSize => {
```

Comment:
I think we're missing a call to `this. cancelSizeListener`, 
We should probably remove the listener at some point

Reply:
at `destroyed` :)